modules/programs/firefox.nix

```diff
@@ -40,6 +60,77 @@ in
         '';
       };
 
+      profiles = mkOption {
+        type = types.attrsOf (types.submodule ({config, name, ...}: {
+          options = {
+            name = mkOption {
+              type = types.str;
+              default = name;
+              description = "Profile name.";
+            };
+
+            id = mkOption {
+              type = types.int;
+              default = 0;
+              description = "Profile ID.";
+            };
+
+            preferences = mkOption {
+              type = types.attrs;
```

Comment:
I believe this should have the type
```
with types; attrsOf (either bool (either int str))
```
for improved configuration merging.

Reply:
I think `types.attrs` should merge just fine: https://github.com/NixOS/nixpkgs/blob/master/lib/types.nix#L226

Reply:
The merging provided by `types.attrs` is very primitive and quite confusing for users in my experience. For example, merging does not take priority into account. So setting the same option in two different places will cause one of them to be chosen more or less randomly. With
```
type = with types; attrsOf (either bool (either int str));
```
one would instead get an error message saying that the option is set twice.